runner/ansible/check.yml

```diff
@@ -35,255 +35,19 @@
-
-    - name: include 1.5.2
-      import_role:
-        name: '1.5.2'
-      when: "'1.5.2' in cluster_selected_checks"
-
-    - name: include 2.1.1
-      import_role:
-        name: '2.1.1'
-      when: "'2.1.1' in cluster_selected_checks"
-
-    - name: include 2.1.2
-      import_role:
-        name: '2.1.2'
-      when: "'2.1.2' in cluster_selected_checks"
-
-    - name: include 2.1.3
-      import_role:
-        name: '2.1.3'
-      when: "'2.1.3' in cluster_selected_checks"
-
-    - name: include 2.1.4
-      import_role:
-        name: '2.1.4'
-      when: "'2.1.4' in cluster_selected_checks"
-
-    - name: include 2.1.5
-      import_role:
-        name: '2.1.5'
-      when: "'2.1.5' in cluster_selected_checks"
-
-    - name: include 2.1.6
-      import_role:
-        name: '2.1.6'
-      when: "'2.1.6' in cluster_selected_checks"
-
-    - name: include 2.1.7
-      import_role:
-        name: '2.1.7'
-      when: "'2.1.7' in cluster_selected_checks"
-
-    - name: include 2.1.8
-      import_role:
-        name: '2.1.8'
-      when: "'2.1.8' in cluster_selected_checks"
-
-    - name: include 2.2.1
-      import_role:
-        name: '2.2.1'
-      when: "'2.2.1' in cluster_selected_checks"
-
-    - name: include 2.2.2
-      import_role:
-        name: '2.2.2'
-      when: "'2.2.2' in cluster_selected_checks"
-
-    - name: include 2.2.3
-      import_role:
-        name: '2.2.3'
-      when: "'2.2.3' in cluster_selected_checks"
-
-    - name: include 2.2.3.exclude
-      import_role:
-        name: '2.2.3.exclude'
-      when: "'2.2.3.exclude' in cluster_selected_checks"
-
-    - name: include 2.2.4
-      import_role:
-        name: '2.2.4'
-      when: "'2.2.4' in cluster_selected_checks"
-
-    - name: include 2.2.5
-      import_role:
-        name: '2.2.5'
-      when: "'2.2.5' in cluster_selected_checks"
-
-    - name: include 2.2.5.exclude
-      import_role:
-        name: '2.2.5.exclude'
-      when: "'2.2.5.exclude' in cluster_selected_checks"
-
-    - name: include 2.2.6
-      import_role:
-        name: '2.2.6'
-      when: "'2.2.6' in cluster_selected_checks"
-
-    - name: include 2.2.7
-      import_role:
-        name: '2.2.7'
-      when: "'2.2.7' in cluster_selected_checks"
-
-    - name: include 2.2.8
-      import_role:
-        name: '2.2.8'
-      when: "'2.2.8' in cluster_selected_checks"
-      
-    - name: include 2.2.9
-      import_role:
-        name: '2.2.9'
-      when: "'2.2.9' in cluster_selected_checks"
-
-    - name: include 2.2.10
-      import_role:
-        name: '2.2.10'
-      when: "'2.2.10' in cluster_selected_checks"
+    - name: Find checks
+      find:
+        paths: "{{ playbook_dir}}/roles/checks"
+        file_type: directory
+      register: checks
+      run_once: yes
+      delegate_to: localhost
+
+    - name: Run check
+      include_role:
+        name: "{{ item.path }}"
+      loop: "{{ checks.files }}"
+      when: "{{ (item.path | basename) in cluster_selected_checks }}"
```

Comment:
Thus use of item here, is conflicting with the use in post results. Also you don't need the "{{ }}" in the when condition Please consider changing to:
    - name: Run check
      include_role:
        name: "{{ CurTest.path }}"
      loop: "{{ checks.files }}"
      loop_control:
        loop_var: CurTest
      when: (CurTest.path | basename) in cluster_selected_checks

Reply:
Yep, already noticed this myself!
Fix is on the way

Reply:
Fixed on: https://github.com/trento-project/trento/pull/255/commits/e54decdc127415b9e2a9c65542c2091395dbd1f6